### ISSUE TITLE
Phishing attack on my electrum wallet 3.3.4

### ISSUE THREAD

**Issue by redtula1488** [spesmilo/electrum#5068]
I wanted to get my private keys generated in a secure bitcoin mixer. But instead, I got a message that I need to download an update for my electrum3.3.4 
Malware was uploaded them here: https://github.com/electrum-official
I didn't know what was going on phishing attack and did not think that the electrum wallet is susceptible to hacker attacks I thought they were guaranteed security and saving my money. But I got a message from their official service. Then I installed it on my computer and wanted to get my private keys, began to import them but nothing happened, the transaction is not displayed, and popped up a message that I closed. Then I decided to check my session in a reliable bitcoin mixer, but found out that my chips were taken away.
I have no information where my private keys are missing, but my coins are not credited to my account. What am I to do? How do I get my funds back? I lost $ 1,000.

**Comment by SomberNight** [comment]
There is no 3.3.4 release.

> Then I installed it on my computer

What you have installed is malware, not Electrum. There is nothing we can do.

> I decided to check my session in a reliable bitcoin mixer, but found out that my chips were taken away

I don't see how this is related.

**Comment by redtula1488** [comment]
> There is no 3.3.4 release.
> 
> > Then I installed it on my computer
> 
> What you have installed is malware, not Electrum. There is nothing we can do.
> 
> > I decided to check my session in a reliable bitcoin mixer, but found out that my chips were taken away
> 
> I don't see how this is related.

I wanted to get the generated private keys into my account. I did not succeed, as offered to download version 3.3.4. After installing this malicious update, I tried to retrieve the generated private keys again and sent them to one of the suggested addresses. But this transaction is notnot displayed and my coins are not credited to my account. After I realized that I had been phished, I removed it and installed version 3.3.3, reinstalled my password. At repeated use of these private keys, writes to me that " inputs are not found"

**Comment by redtula1488** [comment]
Why can't I get my private keys? Writes that the inputs are not found, since they have to be confirmed?

**Comment by SomberNight** [comment]
Are you trying to sweep? Try creating an imported wallet instead.

**Comment by redtula1488** [comment]
> Are you trying to sweep? Try creating an imported wallet instead.

Yeah, I'm trying to sweep, but I can't do anything after a phishing attack on my wallet. I need to pick up my coins from the bitcoin mixer. I swept them when you have inadvertently installed a malicious update to 3.3.4. but they didn't come into my account. Now I reinstalled electrum, changed the password. I'm trying to sweep the private keys generated in the bitcoin mixer, but electrum writes to me that the inputs are not found, as they need to be confirmed. What's the meaning of that?

**Comment by SomberNight** [comment]
It means that at least according to the server you are connected to, those keys don't have any unspent coins. You should create an imported wallet, as I has said, to see their history.

**Comment by redtula1488** [comment]
But why do these keys have no coins if I generated them in bitcoin mixer? The money was stolen by cybercriminals? How do I create an imported wallet?

**Comment by redtula1488** [comment]
> It means that at least according to the server you are connected to, those keys don't have any unspent coins. You should create an imported wallet, as I has said, to see their history.

I figured out how to import the wallet, did it. Now I see the history of these keys, but where did my coins go? They were stolen by intruders during a phishing attack, because I installed a malicious virus or I can return them someow?

**Comment by redtula1488** [comment]
**I created an imported wallet and looked at the history of my private keys, as you told me. They were withdrawn to an unknown account. How did this happen? What to do now?**

**Comment by gits7r** [comment]
We are sorry for this, but this message is confusing and too alarming and causes panic among users.

Electrum doesn't have a bug that can be exploited, it cannot be controlled remotely, it has no open vulnerability that can cause loss without user's action. Electrum was no more "hacked" or "exploited" than gmail, yahoo, outlook and all financial institutions (banks, etc.) as well as various other online services are every day.

Because of how peer discovery works in Electrum, there is not much we can do for old versions, since we can't prevent them with 100% success rate to run into a malicious server. This is because, unlike other lightweight wallets, Electrum decided to not have only few harcoded servers that will be responsible for the privacy of all users, and act as single point of failure, but instead allow users to run their own servers or use servers that they trust. Electrum takes user privacy very seriously, which is why proper peer to peer discovery without central authority arbitration was adopted, instead of anything else. This way an attacker cannot keep an Electrum user offline, or isolate him, or pull various attacks.

While the entire Electrum team is doing absolutely everything possible to protect the users, such as:

- patch Electrum wallet to not display rich text, and don't allow arbitrary messages, only strict codes;

- patch ElectrumX server implementation to detect sybil (malicious servers that send the phishing message) and not further broadcast them to clients;

- implement blacklist logic to maintain malicious servers outside the view of the clients;

- heavily advertise on social, website and all communication forms existent with the users that they should always run the latest version and always only install from the official source (electrum.org), accessed over secure protocol (https) with prior verifications of the PGP signature;

...the sad truth is that nothing can be truly done to protect an user from its own actions. If you are willing to install Electrum from a different source, when the official is electrum.org, and you don't verify signatures, even with the latest patch that does not display rich text you are still vulnerable as you can receive an email or text message with the same phishing message, and install a backdoored Electrum.

After all, when you install and use security software and finances software such as Electrum the first rule is to make sure you are running a version that has no discovered vulnerabilities and your build is signed and genuine.

I know this is not pleasant to read after loss of funds, and we are sorry, but this is the sad truth. This is not a vulnerability in Electrum, so we are going to respectfully close such issues / tickets on github because we are already doing everything possible to limit the effects of phishing attacks, and such issues do not provide any new information.

**Comment by redtula1488** [comment]
**Is it really possible that while trying to sweep the electrum private keys generated in the bitcoin mixer, they were able to steal my coins? They went to the wrong address, I specified in electrium, and redirect to another account.**

**Comment by gits7r** [comment]
It's not possible to tell what might have happened given we can't look at the code of Electrum backdoored version used by you.

There is no official Electrum 3.3.4, so you must used something fishy. As long as you use a backdoor version, the answer if yes, if the malware is coded to steal sweep private keys, it does so. Most probably what happened is that the malware is simpler, and it just replaces the destination address with the attacker's address. Sorry.

**Comment by redtula1488** [comment]
> It's not possible to tell what might have happened given we can't look at the code of Electrum backdoored version used by you.
> 
> There is no official Electrum 3.3.4, so you must used something fishy. As long as you use a backdoor version, the answer if yes, if the malware is coded to steal sweep private keys, it does so. Most probably what happened is that the malware is simpler, and it just replaces the destination address with the attacker's address. Sorry.

I removed electrum 3.3.4 malware from my system and installed the client from the official 3.3.3 website. Can be considered safe and continue to use? What additional measures should be taken to prevent further data theft? I am very shocked by what happened and I am afraid that there may be some additional theft of my personal data

**Comment by gits7r** [comment]
In order to be perfectly safe, I recommend you do a fresh install from scratch of the operating system. From 0.

We do not know what this malware might have had in it and it's hard to say.

Install from scratch, after that download 3.3.3 from https://electrum.org (check for https://electrum.org in the address bar), create a new seed (new wallet entirely), back that seed up and continue to use it as until now.

if you reach a server that doesn't broadcast your transaction and replies with the phishing message, just use a different server.

It's not the end of the world, you were just lead into a confusion by the phishing message. It's the oldest scam in the history of the internet and unfortunately it still leads many many people into confusions every day  (Not only in Electrum but also in all other online services / banks  / emails, etc.). Keep on playing, but more attention in the future about sources of download of security software and finance software like Electrum.

**Comment by redtula1488** [comment]
This is not the end, but nevertheless most of my savings have been lost and cannot be recovered. Since I did not go to the official website of electrum, so I have not had the opportunity to see the notice about phishing attacks, and other ways to notify their customers of electrium not found, although I regularly used it recently, so I and many others were introduced so easily mislead, I did not even know that Electrium I can attackers send vredonostnogo links.Until recently, it seemed to me that any alerts can send only the official service, so I was so trusting to this message with malware. Sadly, I feel like a complete fool.

**Comment by MyLifeIsOver** [comment]
> I wanted to get my private keys generated in a secure bitcoin mixer. But instead, I got a message that I need to download an update for my electrum3.3.4
> Malware was uploaded them here: https://github.com/electrum-official
> I didn't know what was going on phishing attack and did not think that the electrum wallet is susceptible to hacker attacks I thought they were guaranteed security and saving my money. But I got a message from their official service. Then I installed it on my computer and wanted to get my private keys, began to import them but nothing happened, the transaction is not displayed, and popped up a message that I closed. Then I decided to check my session in a reliable bitcoin mixer, but found out that my chips were taken away.
> I have no information where my private keys are missing, but my coins are not credited to my account. What am I to do? How do I get my funds back? I lost $ 1,000.

I was robbed of a very high value, which was years ago to pay for a medical treatment, I am dying slowly. so I still have hope because I have God in my heart.
I created a new account and I am accepting donations to anyone who feels the desire to help me.
Thank you and best wishes to you and your family.

My bitcoin Wallet:  1FcvMGQN9c5aE1o3WQnFqhUN9ShQssNoYd

**Comment by AbdussamadA** [comment]
@redtula1488 the 3.3.4 wallet was malware. as soon as you exposed the private keys to it your coins were stolen.

Before using this system to store bitcoins again you should reformat your hard drive and reinstall your OS. That is the only way to be sure you are rid of the malware.
